contracts/src/v0.8/automation/dev/v2_3/AutomationRegistrar2_3.sol

```diff
@@ -437,19 +365,20 @@ contract AutomationRegistrar2_3 is TypeAndVersionInterface, ConfirmedOwner, IERC
    * @dev register upkeep on AutomationRegistry contract and emit RegistrationApproved event
    */
   function _approve(RegistrationParams memory params, bytes32 hash) private returns (uint256) {
-    IAutomationRegistryMaster2_3 AutomationRegistry = s_config.AutomationRegistry;
-    uint256 upkeepId = AutomationRegistry.registerUpkeep(
+    IAutomationRegistryMaster2_3 registry = s_registry;
+    uint256 upkeepId = registry.registerUpkeep(
       params.upkeepContract,
       params.gasLimit,
       params.adminAddress,
       params.triggerType,
+      address(params.billingToken), // have to cast as address because master interface doesn't use contract types
       params.checkData,
       params.triggerConfig,
       params.offchainConfig
     );
-    bool success = LINK.transferAndCall(address(AutomationRegistry), params.amount, abi.encode(upkeepId));
+    bool success = LINK.transferAndCall(address(registry), params.amount, abi.encode(upkeepId));
```

Comment:
nit: I feel that having LINK here is a bit confusing, how about adding a comment to explain that it will invoke the `onTokenTransfer` on registry, and this is not limited to LINK. 

OTOH, if we dont use `IERC677Receiver`, I guess we can just have sth like `acceptRegistration()` in registry to accept the fund and update upkeep. With this, we get rid of LINK in code, but I guess this is too much overhead for now. WDYT?

Reply:
ah shoot this is an area I need to work on also. Good catch! Again, need better tests...